homeassistant/util/unit_conversion.py

```diff
@@ -330,40 +308,32 @@ def convert(
         )
 
     @classmethod
-    def _fahrenheit_to_celsius(cls, fahrenheit: float, interval: bool = False) -> float:
+    def convert_interval(cls, value: float, from_unit: str, to_unit: str) -> float:
+        """Convert a temperature interval from one unit to another."""
+        return super().convert(value, from_unit, to_unit)
```

Comment:
This is the main change in this PR.
Instead of having `interval` as an argument, it moves it to a separate function, which I think it makes more sense than having it as an argument.

Since TemperatureConverter is new in 2022.10, then if it is approved I think it should be added to beta.